cpp/oneapi/dal/algo/knn/backend/gpu/train_kernel_brute_force_dpc.cpp

```diff
@@ -78,7 +79,10 @@ static train_result<task::classification> call_daal_kernel(const context_gpu& ct
     // are not modified by the algorithm.
     const bool copy_data_responses = false;
     knn_model->impl()->setData<Float>(daal_data, copy_data_responses);
-    knn_model->impl()->setLabels<Float>(daal_responses, copy_data_responses);
+    if constexpr (!std::is_same_v<Task, task::search>) {
+        daal_responses = interop::convert_to_daal_table<Float>(responses);
```

Comment:
please use dpc converter with queue:
interop::convert_to_daal_table(queue, responses)

Reply:
fixed

Reply:
as i remember it is not a template func, please remove <Float> also